plugin/src/main/java/org/openrewrite/gradle/RewritePlugin.java

```diff
@@ -80,9 +80,18 @@ public void apply(Project rootProject) {
                 if(!(plugin instanceof JavaBasePlugin)) {
                     return;
                 }
-                JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
-                javaConvention.getSourceSets().all(sourceSet -> {
-                    sourceSets.add(sourceSet);
+                //Collect Java metadata for each project (used for Java Provenance)
+                JavaPluginExtension javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
```

Comment:
`JavaPluginExtension` was added in Gradle 4.10, which is newer than the oldest version we currently support (4.7), and we were talking about going back to support even earlier versions of Gradle. Ultimately to support Gradle versions new and old we'll need to use reflection to load one or the other at runtime. For now stick with `JavaPluginConvention` if you can.

Reply:
makes sense